packages/smooth_app/lib/helpers/text_input_formatters_helper.dart

```diff
@@ -10,14 +10,28 @@ import 'package:smooth_app/helpers/strings_helper.dart';
 /// on the user's language.
 ///
 /// It works if there are no group separator in the format, as it would be
-/// problematic with the `123,456.9` and both comma and dot.
+/// problematic with the `1,234,567.89` and both comma and dot.
 ///
 /// Also, if a separator is already displayed, it will be moved to the new
 /// position
 class DecimalSeparatorRewriter extends TextInputFormatter {
   DecimalSeparatorRewriter(NumberFormat format)
       : _decimalSeparator = format.symbols.DECIMAL_SEP,
-        _separatorToReplace = _findSeparatorToReplace(format);
+        _separatorToReplace = _findSeparatorToReplace(format) {
+    // Here we check that there are no group separators.
+    // The formatted string should
+    // * contain the nine digits 1..9
+    // * contain one decimal separator (either ',' or '.')
+    // * have a length of 10 (9 digits and one separator)
+    const double number = 1234567.89;
+    const String formattedDot = '1234567.89';
+    const String formattedComma = '1234567,89';
+    final String formatted = format.format(number);
+    if (formatted != formattedDot && formatted != formattedComma) {
+      throw Exception(
+          'Wrong format: $formatted found, either $formattedDot or $formattedComma expected');
+    }
```

Comment:
This shouldn't happen on any production device, right?

Reply:
@M123-dev I'm afraid it would crash in production env too.
`assert` may make more sense.
I don't how we could prevent users from using "corrupted decimal separator" in a gentle way, like, if they're in a country where the decimal separator is neither dot nor comma.